[CI] SSLErrorMessageFileTests class failing

**Build Scans:**
- [elasticsearch-periodic #6650 / openjdk24_entitlements_true_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/5txda56cxtqgy)
- [elasticsearch-periodic #6650 / openjdk24_entitlements_false_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/xpypggj2iyf4e)
- [elasticsearch-periodic #6643 / openjdk24_entitlements_false_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/6w6mnuh55tsjo)
- [elasticsearch-periodic #6643 / openjdk24_entitlements_true_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/kjd6eofxmrtdc)
- [elasticsearch-periodic #6636 / openjdk24_entitlements_true_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/gimrqmggdbb3i)
- [elasticsearch-periodic #6636 / openjdk24_entitlements_false_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/ac5m4uvn5nj4y)
- [elasticsearch-periodic #6629 / openjdk24_entitlements_false_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/2uipt2rsch6pa)
- [elasticsearch-periodic #6629 / openjdk24_entitlements_true_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/6opkl3nkefowe)
- [elasticsearch-periodic #6622 / openjdk24_entitlements_false_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/uca6hk6wgswmc)
- [elasticsearch-periodic #6622 / openjdk24_entitlements_true_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/7uuw6yijrlpsw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:security:test" --tests "org.elasticsearch.xpack.ssl.SSLErrorMessageFileTests.testMessageForCertificateAuthoritiesOutsideConfigDir" -Dtests.seed=980B56688AF2245A -Dtests.jvm.argline="-Des.entitlements.enabled=true" -Dtests.locale=zgh-Tfng-MA -Dtests.timezone=Asia/Bahrain -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ssl.SSLErrorMessageFileTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.ssl.SSLErrorMessageFileTests),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a throwable with message of "failed to load SSL configuration [xpack.monitoring.exporters.http.ssl] - cannot read configured PEM certificate_authorities [/this/path/is/outside/the/config/directory/file.error] because access to read the file is blocked; SSL resources should be placed in the [/dev/shm/bk/bk-agent-prod-gcp-1740575026537098452/elastic/elasticsearch-periodic/x-pack/plugin/security/build/testrun/test/temp/org.elasticsearch.xpack.ssl.SSLErrorMessageFileTests_980B56688AF2245A-001/tempDir-053/config] directory"
     but: was <org.elasticsearch.ElasticsearchSecurityException: failed to load SSL configuration [xpack.monitoring.exporters.http.ssl] - cannot read configured PEM certificate_authorities [/this/path/is/outside/the/config/directory/file.error] because the file does not exist> at org.elasticsearch.xpack.core.ssl.SSLService.lambda$loadSslConfigurations$11(SSLService.java:622)
```

**Issue Reasons:**
- [main] 17 consecutive failures in step openjdk24_entitlements_true_checkpart4_java-matrix
- [main] 17 consecutive failures in step openjdk24_entitlements_false_checkpart4_java-matrix
- [main] 34 failures in class org.elasticsearch.xpack.ssl.SSLErrorMessageFileTests (3.5% fail rate in 958 executions)
- [main] 17 failures in step openjdk24_entitlements_true_checkpart4_java-matrix (100.0% fail rate in 17 executions)
- [main] 17 failures in step openjdk24_entitlements_false_checkpart4_java-matrix (100.0% fail rate in 17 executions)
- [main] 17 failures in pipeline elasticsearch-periodic (100.0% fail rate in 17 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageFileTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageFileTests.java
@@ -16,6 +16,7 @@ import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.PathUtils;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.jdk.RuntimeVersionFeature;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.xpack.core.ssl.SSLService;
 import org.junit.Before;
@@ -362,6 +363,11 @@ public class SSLErrorMessageFileTests extends ESTestCase {
         String configKey,
         BiConsumer<String, Settings.Builder> configure
     ) throws Exception {
+        assumeTrue(
+            "Requires Security Manager to block access, entitlements are not checked for unit tests",
+            RuntimeVersionFeature.isSecurityManagerAvailable()
+        );
+
         final String prefix = randomSslPrefix();
         final Settings.Builder settings = Settings.builder();
         configure.accept(prefix, settings);

